messaging/src/main/java/org/axonframework/commandhandling/gateway/DefaultCommandGateway.java

```diff
@@ -118,4 +156,29 @@ private <C> CommandMessage<C> asCommandMessage(Object command, MetaData metaData
                 metaData
         );
     }
+
+    private <C> CommandMessage<C> enrichCommandMessage(CommandMessage<C> commandMessage) {
+        if(routingKeyResolver == null && priorityCalculator == null) {
```

Comment:
Is it possible that just one of the resolvers is not null? Or we always require routingKey with priority? Is it possible to have routingKey without priority and vice versa? If not I may think about coupling it into one object. 

For me this check looks redundant, we already have checks in resolveRoutingKey / resolvePriority

Reply:
Yes, it's possible.
This check is to prevent the creation of a new message it not necessary. If both are not present, what the point of creating a second message, with the priority and routingkey still empty?